common/fixation_entry.go

```diff
@@ -230,13 +241,22 @@ func (fs *FixationStore) getUnmarshaledEntryForBlock(ctx sdk.Context, index stri
 	// entry with block 199 would return the entry with block 100, which was in
 	// effect at the time of block 199.
 
+	latest := true
+
 	for ; iterator.Valid(); iterator.Next() {
 		var entry types.Entry
 		fs.cdc.MustUnmarshal(iterator.Value(), &entry)
 
 		if entry.GetBlock() <= block {
+			// TODO: remove exception for "latest" when latest gets refcount
+			if !latest && entry.IsStale(ctx) {
```

Comment:
just return from the function, "latest " is not needed

Reply:
Indeed `latest` is not needed - done.
The `break` is useful to ensure one path for "not found".